src/main/java/com/arpnetworking/metrics/mad/parsers/CollectdJsonToRecordParser.java

```diff
@@ -0,0 +1,389 @@
+                final String type = record.getType();
+                final String typeInstance = record.getTypeInstance();
+
+                for (final CollectdRecord.Sample sample : record.getSamples()) {
+                    final String metricName = computeMetricName(plugin, pluginInstance, type, typeInstance, sample.getDsName());
+                    final MetricType metricType = mapDsType(sample.getDsType());;
+                    final Metric metric = new DefaultMetric.Builder()
+                            .setType(metricType)
+                            .setValues(Collections.singletonList(
+                                    new Quantity.Builder().setValue(sample.getValue()).build()))
+                            .build();
+                    metrics.merge(metricName, metric, CollectdJsonToRecordParser::mergeMetrics);
+                }
+                builder.setMetrics(metrics);
+                builders.add(builder);
+            }
+            return builders;
+        } catch (final IOException ex) {
+            // CHECKSTYLE.OFF: IllegalInstantiation - Approved for byte[] to String
+            throw new ParsingException(
+                    String.format("Error parsing collectd json; data=%s", new String(data, Charsets.UTF_8)),
+                    ex);
+            // CHECKSTYLE.ON: IllegalInstantiation
+        }
+    }
+
+    private String computeMetricName(
+            final String plugin,
+            final String pluginInstance,
+            final String type,
+            final String typeInstance,
+            final String dsName) {
+        final StringBuilder builder = new StringBuilder();
+        builder.append(plugin);
+        if (pluginInstance != null && pluginInstance.length() > 0) {
+            builder.append("/");
+            builder.append(pluginInstance);
+        }
+        builder.append("/");
+        builder.append(type);
+        if (typeInstance != null && typeInstance.length() > 0) {
+            builder.append("/");
+            builder.append(typeInstance);
+        }
+        if (dsName != null && dsName.length() > 0) {
+            builder.append("/");
+            builder.append(dsName);
+        }
+        return builder.toString();
+    }
+
+    private MetricType mapDsType(final String type) {
+        switch (type) {
+            case "gauge":
+                return MetricType.GAUGE;
+            case "absolute":
+                // This is an odd type.  It is a counter that is reset on read and divided by the last time.
+                return MetricType.COUNTER;
+            case "counter":
+                return MetricType.COUNTER;
+            case "derive":
+                return MetricType.COUNTER;
+            default:
+                return MetricType.GAUGE;
+        }
+    }
+
+    private static Metric mergeMetrics(final Metric m1, final Metric m2) {
+        final ArrayList<Quantity> combinedValues = Lists.newArrayList(m1.getValues());
```

Comment:
This will be slow as hell. It would be better to collect them into a multi-map type structure and then collapse. No?